src/context/node.js

```diff
@@ -274,11 +293,38 @@ export class ContextNode {
   }
 
   /**
-   * @return {boolean}
-   * @protected Used cross-binary.
+   * Add or update a component with a specified ID. If component doesn't
+   * yet exist, it will be created using the specified factory. The use
+   * of factory is important to reduce bundling costs for context node.
+   *
+   * @param {*} id
```

Comment:
Why can `id` be any type? The tests for this appear to always give it a `!Node`

Reply:
It's most often the component function itself. But sometimes it's better to map it to something else. For instance, the subscriber case is modeled as a component that calls the subscriber handler. The component itself is a fully reusable function, but subscriber handlers are unique to the caller. So it's used as an `id`. Another example could provide an id as a globally unique string. I don't overall anticipate a lot of need for cross-binary components. But I believe some will be. In one of the previous cases, @samouri suggested that we could use components as property setters. In that case, it'd be better to use the property's object (or key) as an ID of the component.